client/packages/ecore-property-view/src/browser/properties-service.ts

```diff
@@ -0,0 +1,378 @@
+    }
+
+    protected getSelectionData(selection: GlspSelection): any { // WorkflowElementSelectionData {
+        if (selection.additionalSelectionData) {
+            return selection.additionalSelectionData.selectionDataMap.get(this.getElementId(selection));
+        }
+    }
+
+    async getProperties(selection: any): Promise<any> {
+        if (selection.selectedElementsIDs) {
+            const selectionData = this.getSelectionData(selection);
+            return this.fetchElement(selectionData.modelUri, selectionData.semanticUri);
+        }
+        return undefined;
+    }
+
+    protected fetchElement(modelUri: string, elementId: string): Promise<string> {
+        return this.modelServerClient.getElementById(modelUri, elementId)
+            .then(response => {
+                const returnObject = response.body as any;
+                // add semanticUri to jsonforms data structure
+                returnObject["semanticUri"] = elementId;
+                return returnObject;
+            });
+    }
+
+    async getSchema(selection: any, properties?: any): Promise<JsonSchema | undefined> {
+        const eClassName = this.getSelectionData(selection).eClass || properties.eClass.split("#//")[1];
+        // FIXME atm we use a local version of the typeschema, as there are performace issues with the whole ecore type schema
+        // const ecoreTypeSchema = await this.getTypeSchema();
+        const ecoreTypeSchema = this.ecoreTypeSchema;
+        if (ecoreTypeSchema.definitions) {
+            /* @ts-ignore */
+            const elementSchema = this.ecoreTypeSchema.definitions[eClassName.toLowerCase()];
+            return {
+                definitions: ecoreTypeSchema.definitions,
+                ...elementSchema
+            };
+        }
+        return undefined;
+    }
+
+    protected async getTypeSchema(): Promise<JsonSchema7> {
+        return this.modelServerClient.getTypeSchema(this.ecoreUri)
+            .then(response => {
+                const typeSchema = response.body as JsonSchema7;
+                // add semanticuri to all definitions
+                if (typeSchema.definitions) {
+                    Object.entries(typeSchema.definitions).forEach(definition => {
+                        if (definition[1].properties) {
+                            definition[1].properties["semanticUri"] = { "type": "string" };
+                        }
+                    });
+                }
+                return typeSchema;
+            });
+    }
+
+    async getUiSchema(selection: any, properties?: any): Promise<UISchemaElement | undefined> {
+        if (properties && properties.eClass) {
+            const eClassName = this.getSelectionData(selection).eClass || properties.eClass.split("#//")[1];
+            return this.modelServerClient.getUiSchema(eClassName.toLowerCase()).then((response: any) => response.body as UISchemaElement);
+        }
+        return undefined;
+    }
+
+    protected ecoreTypeSchema = {
```

Comment:
Can we move this schema into its own *.json file?

Reply:
I moved it temporarily to the new common package, such that both the property view and the tree editor can use it. It is already available to fetch via the modelserver, but I noticed severe performance issues when the complete schema is fetched. Therefore we use temporarily this subset of the schema locally.